waltz-client/src/main/java/com/wepay/waltz/client/internal/Partition.java

```diff
@@ -683,4 +718,35 @@ public String toString() {
 
     }
 
+    private Boolean isMounted() {
+        synchronized (lock) {
```

Comment:
Is there a reason to use sync here? `mounted` is `volatile boolean`. If you are just observing for metrics, I think sync is unnecessary.

Reply:
sorry, my bad. completely missed to notice that the variable is declared as volatile. Will fix it.